JobConfig/primary/prolog.fcl

```diff
@@ -16,59 +15,75 @@ Primary: {
     g4run : @local::mu2e.physics.producers.g4run.muons
     genCounter: { module_type: GenEventCounter }
     @table::EventGenerator.producers
-    @table::CommonMC.producers
-    @table::TrackerMC.producers
-    @table::CaloDigiMC.producers
-    @table::CrvDAQPackage.producers
-# dedicated digi maker for filter
-# 2 stages now
-    makeSGS : {
-      module_type : MakeStrawGasSteps
-      KeepDeltasModule : "g4run" # don'e compress deltas on primary particles
+    @table::TrackerMC.StepProducers
+    @table::CaloMC.StepProducers
+    # the following should be reconfigured to run after step compression once that is working
+    FindMCPrimary : {
+      module_type : FindMCPrimary
+      SinglePrimary : true
+      GenParticles : "generate"
+      SimParticles : "g4run"
+      PrimaryGenIds : [] # this is intentionally empty: there should only be 1 GenParticle in primary production
     }
-    makeSD:{
-      @table::makeSD
-    }
-    compressDigiMCs : @local::DigiCompression.Primary
   }
   filters : {
 # generator truncation filter; only needed for cosmic rays in principle
     g4status: {
       module_type: FilterStatusG4
       input: "g4run"
     }
-# digi filtering for Primary selection.
-    DigiFilter : {
-      module_type : StrawDigiMCFilter
-      StrawDigiMCCollection : "makeSD"
-    }
-# shower filter for calorimeter selection
-    CaloFilter : {
-      module_type : CaloShowerSimFilter
+# step filter for Sim Primary selection
+    DetStepFilter : {
+      module_type : DetectorStepFilter
+      StrawGasSteps : [ "StrawGasStepMaker" ] 
+      CaloShowerSteps : [ "CaloShowerStepMaker" ]
+      CrvSteps : [ "g4run:CRV" ]
+      MinimumTrkStepEnergy : 1.0e-6 # 1 ev
+      MinimumCaloStepEnergy : 0.01 # need a realistic value FIXME!
+      MinimumCrvStepEnergy : 1.0e-6 # get a real value from CRV experts FIXME!
+      MinimumPartMom : 80.0 # MeV/c
+      MaximumPartMom : 10000.0 # MeV/c
+      KeepPDG : [-11, 11, -13, 13 ] # electrons and muons: should be defined in a common config file FIXME
```

Comment:
You might want to add pions to the list, but probably their contribution is negligible.

Reply:
Are you thinking of pions from cosmic rays that directly create a track or calo cluster?  I can add them, it can't really hurt.  Note that special primaries (protons, photons) override this as appropriate.

Reply:
Yes, pions from cosmics.